### PR TITLE
CDAP-6455 Added cdap-common.jar at start in the Yarn application classpath for MR launched by explore.

### PR DESCRIPTION
**Pull request by sagarkapare**

### CODE REVIEW

**Comment by sagarkapare** [comment]
Verified on cluster that explore queries run successfully.

File: cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java [hunk 1]
```diff
@@ -453,11 +469,15 @@ private static File updateYarnConfFile(File confFile, File tempDir) {
     String yarnAppClassPath = conf.get(YarnConfiguration.YARN_APPLICATION_CLASSPATH,
                                        Joiner.on(",").join(YarnConfiguration.DEFAULT_YARN_APPLICATION_CLASSPATH));
 
-    // add the pwd/* at the beginning of classpath. so user's jar will take precedence and without this change,
-    // job.jar will be at the beginning of the classpath, since job.jar has old guava version classes,
-    // we want to add pwd/* before
-    yarnAppClassPath = "$PWD/*," + yarnAppClassPath;
+    // We first add cdap-common.jar to the classpath so that FileContextLocationFactory from cdap-common gets
+    // picked instead of FileContextLocationFactory from Apache Twill. We then add the pwd/* to the classpath.
+    // so user's jar will take precedence. Without this change, job.jar will be at the beginning of the classpath.
```
Comment (poornachandra) [inline cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java]: `Without this change,...` =>  `Without pwd/* in the beginning of classpath,...`

**Comment by poornachandra** [comment]
LGTM

File: cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java (hunk 1 above)
Comment (sagarkapare) [inline cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java]: fixed.

**Comment by poornachandra** [comment]
LGTM

**Comment by sagarkapare** [comment]
Build is running here - http://builds.cask.co/browse/CDAP-DUT4493-3

**Comment by sagarkapare** [comment]
Build is green here http://builds.cask.co/browse/CDAP-DUT4493-4 and verified on cluster that Explore queries are succeeding. Merging this PR.

**Pull request merged**